src/coreclr/src/gc/gc.cpp

```diff
@@ -9958,7 +9988,8 @@ HRESULT gc_heap::initialize_gc (size_t soh_segment_size,
         check_commit_cs.Initialize();
     }
 
-    if (!reserve_initial_memory (soh_segment_size, loh_segment_size, poh_segment_size, number_of_heaps, use_large_pages_p))
+    bool pohExplicitlyZeroed = (heap_hard_limit_soh) && (GCConfig::GetGCHeapHardLimitPOH() == 0) && (GCConfig::GetGCHeapHardLimitPOHPercent() == 0);
```

Comment:
>pohExplicitlyZeroed [](start = 9, length = 19)

nit - 
`pohExplicitlyZeroed`
-> something like 
`zero_poh_requested_p`

aside from casing, pohExplicitlyZeroed also sounds like you want to zero the memory on POH in the GC context.